lib/plugins/aws/package/compile/events/apiGateway/index.js

```diff
@@ -20,15 +21,261 @@ const getMethodAuthorization = require('./lib/method/authorization');
+            ],
+          },
+        },
+        additionalProperties: false,
+      },
+    },
+  },
+  additionalProperties: false,
+};
+
+const baseIntegrationSchema = {
+  type: 'object',
+  properties: {
+    cors: corsSchema,
+    method: {
+      enum: extendWithUppercase(httpMethods),
+    },
+    path: { type: 'string' },
+    operationId: { type: 'string' },
+    async: { type: 'boolean' },
+    private: { type: 'boolean' },
+    request: requestSchema,
+    response: responseSchema,
+    authorizer: authorizerSchema,
+  },
+  required: ['path'],
+  additionalProperties: false,
+};
+
+const lambdaProxyIntegrationSchema = _.merge({}, baseIntegrationSchema, {
+  properties: {
+    integration: {
+      enum: extendWithUppercase(['lambda_proxy']),
+    },
+  },
+  required: ['integration'],
+});
+
+const mockIntegrationSchema = _.merge({}, baseIntegrationSchema, {
+  properties: {
+    integration: {
+      enum: extendWithUppercase(['mock']),
+    },
+  },
+  required: ['integration'],
+});
+
+const awsIntegrationSchema = _.merge({}, baseIntegrationSchema, {
+  properties: {
+    integration: {
+      enum: extendWithUppercase(['aws', 'aws_proxy']),
+    },
+  },
+  required: ['integration'],
+});
+
+const httpIntegrationSchema = _.merge({}, baseIntegrationSchema, {
+  properties: {
+    integration: {
+      enum: extendWithUppercase(['http', 'http_proxy']),
+    },
+    request: {
+      required: ['uri'],
+    },
+  },
+  required: ['integration'],
+});
+
+const httpVpcLinkIntegrationSchema = _.merge({}, baseIntegrationSchema, {
+  properties: {
+    connectionType: { enum: ['vpc-link', 'VPC_LINK'] },
+    connectionId: { type: 'string' },
+    integration: {
+      enum: extendWithUppercase(['http', 'http_proxy']),
+    },
+    request: {
+      required: ['uri'],
+    },
+  },
+  required: ['integration', 'connectionType', 'connectionId'],
+});
+
+const lambdaIntegrationSchema = _.merge({}, baseIntegrationSchema, {
+  properties: {
+    integration: {
+      enum: extendWithUppercase(['lambda']),
+    },
+    authorizer: {
+      properties: {
+        claims: { type: 'array', items: { type: 'string' } },
+      },
+    },
+  },
+  required: ['integration'],
+});
+
+const cognitoIntegrationSchema = _.merge({}, baseIntegrationSchema, {
+  properties: {
+    integration: {
+      enum: extendWithUppercase(['cognito_user_pools']),
+    },
+    authorizer: {
+      required: ['name'],
+    },
+  },
+  required: ['integration'],
+});
+
 class AwsCompileApigEvents {
   constructor(serverless, options) {
     this.serverless = serverless;
     this.options = options;
     this.provider = this.serverless.getProvider('aws');
 
     this.serverless.configSchemaHandler.defineFunctionEvent('aws', 'http', {
-      // TODO: Complete schema, see https://github.com/serverless/serverless/issues/8018
-      anyOf: [{ type: 'string' }, { type: 'object' }],
+      oneOf: [
+        { type: 'string', pattern: `^(${httpMethods.join('|')}) \\S+` },
```

Comment:
I think we need to use `regex` to be case insensitive (same as we did in HTTP API: https://github.com/serverless/serverless/blob/5851bcadcf6016914013d447de899cc98efb8346/lib/plugins/aws/package/compile/events/httpApi/index.js#L57)

Note that supporting just all lower case and all upper case will probably be not good enoug, as I take `Get` or `Post` also works

Reply:
Do you think we should have the same case-insensitive treatment (using regex) for similar cases like `authorizer.type` and `integration`?

Reply:
I think whenever we need to use regexp pattern and expect case insensitive matching, we should use `regex` instead of `pattern`

So if it's the case for `authorizer.type` an `integration` then, yes